### ISSUE TITLE
Hashtag tab completion, render hashtags as links which open search box

### ISSUE THREAD

**Issue by cspotcode** [stevencohn/OneMore#1514]
*I put two features into this issue, because implementing one makes it easier to implement the other.  But I would be happy to split them into 2 issues if you prefer*

# Problem to Solve

Save typing, avoid typos, avoid the need to memorize hashtags.

Smooth navigation between all pages linked by a hashtag: click the tag in one place, it pops up the search box linking to all other occurrences of that hashtag.

# New Feature/Solution

Type `#`, optionally some characters to narrow the search, then press Ctrl+Space.  A dropdown appears which uses fuzzy search to narrow down a list of all known hashtags.  Selecting a hashtag from the list with Arrow keys and Enter will insert it into the document.

For example, supposing I have hashtags for learning a new language: #japanese, #japanese-conjugation, #japanese-phrases, etc.  Typing `#jc`, ctrl-space, and the fuzzy matcher shows me a dropdown of all hashtags which contain the characters `jc` in that order.  It suggests #japanese-conjugation, which I can choose with arrows and Enter.

---

Because the user interacts with the dropdown, and it's modifying the document, it can also render the hashtag as a clickable link which shows the hashtag search dialog.

## Alternative Solutions

I tried to find other tab-completion solutions I could use to achieve this.  But I couldn't find anything in OneNote vanilla nor OneMore that offer a selectable list of options.  Snippets, as far as I can tell, require you to type the name correctly from memory.

Rendering hashtags as clickable links could also be implemented as a post-process plugin script.  Is there a URI that triggers the hashtag search box?

I also tested placing the caret inside a hashtag, then pressing Alt-F9 hotkey to open the hashtag search.  But it doesn't prepopulate the search box with the hashtag.  That could be another option.

# Additional Context

**Comment by cspotcode** [comment]
I can't remember if I was aware of the Alt+T hashtag picker when I filed this issue.  It seems really close to offering hashtag completions, if it somehow supported keyboard-only input.  Say, if it auto-focused the input box, and let me type a few characters and press enter to choose the matced hashtag.

**Comment by stevencohn** [comment]
It's in the plans, but this was an initial release with. I prefer incremental implementation for people to respond an influence the direction, rather than investing too much time in things people don't actually use or enjoy (hence replacing page tags with hashtags! 😄 )

**Comment by stevencohn** [comment]
### Notes

_Insert_ 
* Position text cursor in text, optionally over a word to use as the start of a tag
* Press Alt+D3 to open an in-position drop down hashtag selection box, with typeahead filtering
* Press Enter to accept hashtag, inserting in blank space or replacing selected word
* Press Esc to cancel and return to editor

_Lookup_
* Position text cursor over a hashtag in the text
* Press Shift+Alt+D3
* The Search Hashtags dialog is presented, with the selected hashtag as the default value, and matching pages reported

**Comment by jasonjac2** [comment]
I can't get Alt+G (Or Ctrl+G) to work (7.4B2) my other KB shortcuts work (2 x machines).  Cool feature.  When I invoke with command palette it didn't seem to get focus.  Release notes and command palette don't agree on shortcut.

**Comment by stevencohn** [comment]
I fixed the command palette focusing problem.
`Alt+G` should be the default binding. If you open Settings/Keyboard Shortcuts, does it say `Alt+G`?

<img width="406" height="170" alt="Image" src="https://github.com/user-attachments/assets/e485abc0-5934-4099-a381-7633581abd5b" />

**Comment by jasonjac2** [comment]
It does say Alt+G.
Just to be clear - the focus issue was that the hashtag lookup / selector didn't get focus when I invoked from the command palette.
This is such a good feature BTW.

**Comment by stevencohn** [comment]
Yep, that's was the `1514 steal focus` branch on this ticket just solved.
